src/compiler/transformers/decorators-to-static/convert-static-members.ts

```diff
@@ -0,0 +1,20 @@
+import ts from 'typescript';
+
+/**
+ * Helper function to detect if a class element fits the following criteria:
+ * - It is a property declaration (e.g. `foo`)
+ * - It has an initializer (e.g. `foo *=1*`)
+ * - The property declaration has the `static` modifier on it (e.g. `*static* foo =1`)
+ * @param classElm the class member to test
+ * @returns true if the class member fits the above criteria, false otherwise
+ */
+export const hasStaticInitializerInClass = (classElm: ts.ClassElement): boolean => {
+  return (
+    classElm.kind === ts.SyntaxKind.PropertyDeclaration &&
+    // we use type assertions here as a `filter()` call for narrowing the type is an order of magnitude more
+    // expensive from a time perspective
```

Comment:
does this refer to using one of the `ts.isSomeNodeType` method? (in this case obviously `ts.isPropertyDeclaration`)

that does give us type narrowing afterward if we do:

```suggestion
  return (
    ts.isPropertyDeclaration(classElm) &&
    // we use type assertions here as a `filter()` call for narrowing the type is an order of magnitude more
    // expensive from a time perspective
```

afaict `isPropertyDeclaration` basically just does this same check too, but with the addition that it's return type is `node is PropertyDeclaration`

Reply:
Oooh, yes! That helps us get rid of all those type assertions! Done in c3ab21a4b